lib/buffer.js

```diff
@@ -158,18 +157,16 @@ function assertSize(size) {
  **/
 Buffer.alloc = function(size, fill, encoding) {
   assertSize(size);
-  if (size <= 0)
-    return createBuffer(size);
-  if (fill !== undefined) {
+  if (size > 0 && fill !== undefined) {
```

Comment:
Oh, right. Maybe, but I’d leave that open for another PR, too, especially as it would introduce the subtle change that the return values of `Buffer.alloc()` would share their `buffer` property.